.pytool/Plugin/BuildReportEvaluation/BuildReportQueries.py

```diff
@@ -0,0 +1,265 @@
+## @file BuildReportQueries.yaml
+#
+# Plugin to evaluate a platform's build report file and verify common settings
+#
+#  Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+from edk2toollib.database.edk2_db import Edk2DB
+from buildreport_table import BrComponent, BrPcd, BrLibrary
+import logging
+
+
+def CompiledButButNotIncludedInAnyFV(edk2db: Edk2DB, reportlevel: str) -> int:

```

Comment:
It is common in a platform with various build "flavors" to build drivers that do not end up in the FV to ensure the overall build is not broken due to modifying a common library even though the driver does not get placed in the flash image for the build.

It seems like this is a natural set of "false positives" that would be nice to eliminate given the motivation for this check.

Reply:
There are cases where a platform compiles something, but does not use it intentionally.

This is intended to alert the platform owner that they are doing this, and they should have the ability to ignore the warning message that is generated. Highlighting that this is occurring is the goal, and the platform owner has the ability to ignore the warning message.

Reply:
If it's clear to them what they need to do, that sounds reasonable.

Reply:
Can you add some detail around this in the warning message?